src/devinterp/slt/llc.py

```diff
@@ -6,106 +6,120 @@
             "llc/std": self.llc_std.cpu().numpy().item(),
-            **{f"llc-chain/{i}": self.llc_per_chain[i].cpu().numpy().item() for i in range(self.num_chains)},
+            **{
+                f"llc-chain/{i}": self.llc_per_chain[i].cpu().numpy().item()
+                for i in range(self.num_chains)
+            },
             "loss/trace": self.losses.cpu().numpy(),
         }
-    
-    def __call__(self, chain: int, draw: int, loss: float):
-        self.update(chain, draw, loss)
 
+    def __call__(self, chain: int, draw: int, loss: float, init_loss: float):
+        self.update(chain, draw, loss, init_loss)
 
 
 class OnlineLLCEstimator(SamplerCallback):
     """
-    Callback for estimating the Local Learning Coefficient (LLC) in an online fashion during a sampling process. 
+    Callback for estimating the Local Learning Coefficient (LLC) in an online fashion during a sampling process.
     It calculates LLCs using the same formula as LLCEstimator, but continuously and including means and std across draws (as opposed to just across chains).
 
     Attributes:
         num_chains (int): Number of chains to run. (should be identical to param passed to sample())
         num_draws (int): Number of samples to draw. (should be identical to param passed to sample())
-        n (int): Number of samples used to calculate the LLC.
+        temperature (float): Temperature used to calculate the LLC.
         device (Union[torch.device, str]): Device to perform computations on, e.g., 'cpu' or 'cuda'.
     """
-    def __init__(self, num_chains: int, num_draws: int, n: int, device="cpu"):
+
+    def __init__(
+        self, num_chains: int, num_draws: int, temperature: float, device="cpu"
+    ):
         self.num_chains = num_chains
         self.num_draws = num_draws
 
-        self.losses = torch.zeros((num_chains, num_draws), dtype=torch.float32).to(device)
+        self.losses = torch.zeros((num_chains, num_draws), dtype=torch.float32).to(
+            device
+        )
         self.llcs = torch.zeros((num_chains, num_draws), dtype=torch.float32).to(device)
+        self.moving_avg_llcs = torch.zeros(
+            (num_chains, num_draws), dtype=torch.float32
+        ).to(device)
 
-        self.n = torch.tensor(n, dtype=torch.float32).to(device)
+        self.temperature = torch.tensor(temperature, dtype=torch.float32).to(device)
 
         self.llc_means = torch.tensor(num_draws, dtype=torch.float32).to(device)
         self.llc_stds = torch.tensor(num_draws, dtype=torch.float32).to(device)
 
         self.device = device
 
-    def update(self, chain: int, draw: int, loss: float):
-        self.losses[chain, draw] = loss 
-        init_loss = self.losses[chain, 0]
-
-        if draw == 0:  # TODO: We can probably drop this and it still works (but harder to read)
-            self.llcs[chain, draw] = 0.
+    def update(self, chain: int, draw: int, loss: float, init_loss: float):
+        self.init_loss = init_loss
```

Comment:
not sure if this is accurate since I haven't read the rest of the code yet, but I'm getting the sense that maybe the `init_loss` passed here is computed outside of the estimator (e.g. using the recorded training loss for a checkpoint). if that's the case, maybe it makes sense to pass it during the llc estimator initialization and set it as a class variable there instead of initializing it at the update step?